react/javascript/package.json

```diff
@@ -12,7 +12,7 @@
   "license": "MIT",
   "scripts": {
     "prepare": "make .codegen && npx -y cpy-cli '**/*.scss' '../dist/src' --cwd='src' --parents",
-    "test": "mocha",
+    "test": "mocha --exit",
```

Comment:
The need for the `--exit` option indicates we may have some async work still waiting in the background.
Did you need this even before adding your own tests?

Reply:
Removing `--exit` didn't cause an issue for me locally. Just pushed a commit to remove it, let's see what happens in CI.

Reply:
Without it mocha was hanging for me (when running via `make`). Must be the version of Node I'm using or something else. TBH I didn't investigate whether it was code/tests that I'd added that was causing the hang.

I had a lot of problems generally with building. For instance, after changing code in the react library, running `PACKAGES="messages react html-formatter" LANGUAGES=javascript make` didn't seem to update html-formatter accordingly. I then tried running `make clean` first, but that caused the subsequent build to fail. In the end, my rebuild cycle was:
```
git clean -dfX
PACKAGES="messages react html-formatter" LANGUAGES=javascript make
```
which is somewhat annoying as it meant I had to reinstall NPM packages every time. Dunno what I was doing wrong, if anything.

Reply:
Also, I couldn't find it documented anywhere, but trial and error suggests that the build system specifically requires NPM v7.

Reply:
> Also, I couldn't find it documented anywhere, but trial and error suggests that the build system specifically requires NPM v7.

You're right and we do need to document that. Those changes happened after you first raised this PR but the devex is quite a bit simpler for JS as a result. From the root of the project you can run `npm i && npm run build` and then you should just be able to run tests etc from any module without any extra building/linking/installing.